core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -107,7 +108,13 @@ private String metadataFileLocation(Table table) {
 
   private FileIO fileIO(Table table) {
     if (table.io() instanceof HadoopConfigurable) {
-      ((HadoopConfigurable) table.io()).serializeConfWith(conf -> new SerializableConfiguration(conf)::get);
+      HadoopConfigurable configurableIo = (HadoopConfigurable) table.io();
+      if (configurableIo.getConf().getBoolean(ConfigProperties.CONFIG_SERIALIZATION_DISABLED, false)) {
+        // do not serialize the Configuration object - configs should be set on the deserializer-side
+        configurableIo.serializeConfWith(conf -> new SkipSerializableConfiguration(conf)::get);
+      } else {
+        configurableIo.serializeConfWith(conf -> new SerializableConfiguration(conf)::get);
+      }
```

Comment:
Is it possible to do this in Hive rather than here? Spark uses its own `SerializableConfiguration` that is compatible with Kryo and calls `serializeConfWith` to customize this. I think ideally that is how Hive would work as well. That avoids needing to have a general Hadoop config property. Instead, we can have one just for Hive where it's okay to set things in the Hadoop conf.

Reply:
That's a good idea, thanks @rdblue! I have refactored the code to use a Hive-side config property and call `serializeConfWith` from the Hive storage handler instead. 

The one thing I've noticed and don't really like is that upon using `SerializationUtil.serializeToBase64(table)`, we enter `SerializableTable.copyOf()/SerializableTable.fileIO()`, which would overwrite any potential custom `SerializableSupplier` we might have set earlier for the table via `serializeConfWith`. I wonder if we might want to provide some customizability for the `SerializableTable.copyOf()` in the future. But that could be addressed in a separate PR, if desired.

Reply:
Also setting the configuration with `HadoopFileIO.setConf(Configuration conf)` might lead to a few surprises, as it changes the `SerializableSupplier` implementation behind the scenes.